src/core/server/saved_objects/migrationsv2/integration_tests/migration_from_older_v1.test.ts

```diff
@@ -0,0 +1,226 @@
+        esClient = coreStart.elasticsearch.client.asInternalUser;
+      });
+    return await Promise.all([startEsPromise, startKibanaPromise]);
+  };
+
+  const getExpectedVersionPerType = () =>
+    coreStart.savedObjects
+      .getTypeRegistry()
+      .getAllTypes()
+      .reduce((versionMap, type) => {
+        const { name, migrations, convertToMultiNamespaceTypeVersion } = type;
+        if (migrations || convertToMultiNamespaceTypeVersion) {
+          const migrationsMap = typeof migrations === 'function' ? migrations() : migrations;
+          const migrationsKeys = migrationsMap ? Object.keys(migrationsMap) : [];
+          if (convertToMultiNamespaceTypeVersion) {
+            // Setting this option registers a conversion migration that is reflected in the object's `migrationVersions` field
+            migrationsKeys.push(convertToMultiNamespaceTypeVersion);
+          }
+          const highestVersion = migrationsKeys.sort(Semver.compare).reverse()[0];
+          return {
+            ...versionMap,
+            [name]: highestVersion,
+          };
+        } else {
+          return {
+            ...versionMap,
+            [name]: undefined,
+          };
+        }
+      }, {} as Record<string, string | undefined>);
+
+  const assertMigrationVersion = (
+    doc: SavedObjectsRawDoc,
+    expectedVersions: Record<string, string | undefined>
+  ) => {
+    const migrationVersions = doc._source.migrationVersion;
+    const type = doc._source.type;
+    expect(migrationVersions ? migrationVersions[type] : undefined).toEqual(expectedVersions[type]);
+  };
+
+  const stopServers = async () => {
+    if (root) {
+      await root.shutdown();
+    }
+    if (esServer) {
+      await esServer.stop();
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+  };
+
+  describe('migrating from 7.3.0-xpack which used v1 migrations', () => {
+    const migratedIndex = `.kibana_${kibanaVersion}_001`;
+
+    beforeAll(async () => {
+      await removeLogFile();
+      await startServers({
+        oss: false,
+        dataArchive: Path.join(__dirname, 'archives', '7.3.0_xpack_sample_saved_objects.zip'),
+      });
+    });
+
+    afterAll(async () => {
+      await stopServers();
+    });
+
+    it('creates the new index and the correct aliases', async () => {
+      const { body } = await esClient.indices.get(
+        {
+          index: migratedIndex,
+        },
+        { ignore: [404] }
+      );
+
+      const response = body[migratedIndex];
+
+      expect(response).toBeDefined();
+      expect(Object.keys(response.aliases!).sort()).toEqual([
+        '.kibana',
+        `.kibana_${kibanaVersion}`,
+      ]);
+    });
+
+    it('copies all the document of the previous index to the new one', async () => {
+      // After plugins start, some saved objects are deleted/recreated, so we
+      // wait a bit for the count to settle.
+      await new Promise((resolve) => setTimeout(resolve, 5000));
```

Comment:
Is it necessary to fix flakiness? Maybe we can retry the test several times instead of relying on a magic timeout?